### ISSUE TITLE
Rectify model values

### ISSUE THREAD

**Issue by samreid** [phetsims/gravity-and-orbits#302]
From a discussion today, @kathy-phet and @arouinfar and I agreed the model values in the "Model" screen should more closely match the "to scale" screen.  Here's a patch and I'll paste more notes about it later.

<details>

```diff
Index: js/model/ModelSceneFactory.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- js/model/ModelSceneFactory.js	(revision 41f62059ff7750924dc54633471f7e5463e32104)
+++ js/model/ModelSceneFactory.js	(date 1581463667546)
@@ -19,7 +19,6 @@
   // constants
   const SUN_RADIUS_MULTIPLIER = 50; // sun radius multiplier for SunEarthMode and SunEarthMoonMode, tuned by hand
   const EARTH_MOON_RADIUS_MULTIPLIER = 800; // earth and moon radius multiplier for SunEarthMode and SunEarthMoonMode, tuned by hand
-  const EARTH_MASS_SCALE_FACTOR = 10200; // tuned by hand so there are 12 model lunar orbits in one model earth orbit
 
   // in days - actual period is 27.322 days, but this sim's model produces a period of 27.6 days (by inspection)
   const MOON_ORBITAL_PERIOD = 27.6;
@@ -29,13 +28,7 @@
    * balances increased mass and so that forces are 1/2 grid cell in default conditions, hand tuned by checking
    * that reducing the distance by a factor of 2 increases the force arrow by a factor of 4
    */
-  const FORCE_SCALE = 0.573 / EARTH_MASS_SCALE_FACTOR;
-
-  /*
-   * Have to artificially scale up the time readout so that SunEarthMode and SunEarthMoonMode modes have a stable
-   * orbits with correct periods since masses are nonphysical. 365 is days in a year.
-   */
-  const SUN_EARTH_MODE_TIME_SCALE = 365.0 / 334.0;
+  const FORCE_SCALE = 0.573;
 
   /**
    * Convenience function that converts days to seconds, using
@@ -76,9 +69,7 @@
       super();
       this.sun.radius *= SUN_RADIUS_MULTIPLIER;
       this.earth.radius *= EARTH_MOON_RADIUS_MULTIPLIER;
-      this.earth.mass *= EARTH_MASS_SCALE_FACTOR;
       this.forceScale *= FORCE_SCALE;
-      this.timeScale = SUN_EARTH_MODE_TIME_SCALE;
 
       // Sun shouldn't move in model modes
       this.sun.isMovable = false;
@@ -97,12 +88,10 @@
       this.earth.radius *= EARTH_MOON_RADIUS_MULTIPLIER;
       this.moon.radius *= EARTH_MOON_RADIUS_MULTIPLIER;
 
-      this.earth.mass *= EARTH_MASS_SCALE_FACTOR;
-      this.moon.vx *= 21;
+      this.moon.vx *= 21 / Math.sqrt( 10200 ) * 20;
       this.moon.y = this.earth.radius * 1.7;
 
       this.forceScale *= FORCE_SCALE;
-      this.timeScale = SUN_EARTH_MODE_TIME_SCALE;
 
       // Sun shouldn't move in model modes
       this.sun.isMovable = false;

```

</details>

**Comment by samreid** [comment]
My notes from discussion with @kathy-phet and @arouinfar last week:

In "to scale" everything should be accurate.  We would like to unify the "model" values with "to scale" as much as possible, so the main change is the radius of each object.  For instance, we would like to have:
* the earth mass should be the actual earth mass
* To get the right motion for the moon, Make a "fake" mass for the Earth, which is used for computing F=Gm1m2/r^2.  Be clear that it is really "fake".
* On "model" for star/planet/moon, we will have to adjust the moon distance from the earth.  Sun earth distance should be accurate.  

Adjust the force vector scales in the "to scale" screen so they are, by default, 1/2 grid length on default circumstances.  This is likely done via `forceScale`.

**Comment by samreid** [comment]
An alternate proposal which may be easier to implement and have better overall behavior.

* Model will have all of the same physics as the "To Scale" screen.  Masses will all be accurate
* Radii will be different in the model (scaled up so things are large and visible)
* For showing the moon orbiting the earth, use correct "to scale" physics, but then take the angle of the moon relative to the earth, and magnify it.

Thinking of this as a magnification done in the view rather than different physics or different masses (or fake masses) may be better.  I haven't yet thought through all the pros and cons.

**Comment by samreid** [comment]
I'd like to discuss the preceding proposal with @arouinfar and/or @kathy-phet before I start on this issue, as it would dramatically change the direction we proceed.

**Comment by kathy-phet** [comment]
Sam, I am intrigued by this model, but it seems like it would be harder to know how to handle any modified configuration where the moon is moved to a different part of the system. I think for any outside party it would be easier to explain that the physics is all correct if they measure the distances with the tape measure and such and that we just use a fake earth mass (which we can provide) in the modeling of the moon's behavior.

**Comment by samreid** [comment]
The "model" screen currently does not include a measuring tape.  We could add once since some measurements like the "sun to earth" distance would be accurate, but other distances like the "sun radius" would be wrong.  Does the proposal for this issue include adding a measuring tape to the "model" screen?

> it seems like it would be harder to know how to handle any modified configuration where the moon is moved to a different part of the system

This would probably be accomplished by choosing an artificial view scale factor that magnifies the "moon-to-earth" distance in the view.  When dragging the moon, it would be done in this magnified scale, and would update the physical moon accordingly.

Also, if we proceed with the proposal for this issue, wouldn't we need to also provide a fake "sun mass for interaction with moon" so that it is possible to put the moon in an orbit around the sun?

The main point of this new proposal is to be more clear that we aren't changing the physics or physical properties (like fake earth mass), but that we are "zooming in" on each body to make them easier to see.  Perhaps if there was a slider "Magnify Bodies" that the user could drag continuously, to see how the bodies are changing size dynamically, and how it adjusts the moon position accordingly.

**Comment by kathy-phet** [comment]
I don't think we would need a fake "sun mass" at all. If they put it near the sun, that's fine. 

I worry about moving the moon in the zoomed in view. Because then the moon could "look" half way between the earth and the sun, but the model would think that its really far from the sun and close to the earth. So the behavior would be off, visually.

**Comment by samreid** [comment]
Would it help if we depict the moon differently, like a partially transparent image with an arrow or indicator line pointing to the true location?  Or is that too big a step in the wrong direction?

**Comment by kathy-phet** [comment]
I don't see that as useful. Can we try the "fake earth mass only" approach for now, and tune that and see how it goes?

**Comment by samreid** [comment]
Sounds good, I'll proceed with that.

**Comment by samreid** [comment]
I'm still planning to implement the "fake earth mass only" approach, but wanted to write down another idea about the alternate proposal in case we come back to it: we could make the moon non-grabbable in the "model" screen.  You could still end up with odd cases, say, if you drag the earth close to the sun.  But maybe they would be less frequent.

I'm not really sure how finnicky fine-tuning the "fake earth mass" will be, but if we want to guarantee identical orbits in the "model" screen and "to scale" screen, the new proposal in https://github.com/phetsims/gravity-and-orbits/issues/302#issuecomment-586821905 would be the best way to guarantee it.

I'm looking forward to trying the "fake earth mass" strategy, maybe it will be easy and we can analytically compute the fake earth mass instead of having to fine tune it empirically.

**Comment by arouinfar** [comment]
@samreid I agree with @kathy-phet that the "fake earth mass only" approach would be preferable. It sounds like that is what you'll be proceeding with and (for now) you don't need any additional feedback from @kathy-phet or myself.

 > we could make the moon non-grabbable in the "model" screen

I would rather not take away interactivity/flexibility from the simulation.

**Comment by samreid** [comment]
> we could make the moon non-grabbable in the "model" screen

Still thinking through the alternative before I resume work on the "fake earth mass" strategy, in case it doesn't pan out.  When dragging the moon, once it gets far enough from the Earth, it would no longer need that nonlinear mapping and it could just be at the coordinates where it is drawn.  It would only have the "special" nonlinear mapping behavior when it is closer to the earth than the sum of their zoomed-in radii.

**Comment by samreid** [comment]
@kathy-phet said this will not be essential for the upcoming dev version.

**Comment by samreid** [comment]
I had some time while waiting on another issue to be addressed, so I worked on this one.  I published a dev version here: https://phet-dev.colorado.edu/html/gravity-and-orbits/1.2.0-dev.8/phet-io/

Summary of changes:
* All mass values in the Model screen match values in the "To Scale" screen
* The moon is kept in orbit by a "moon adjustment" scale factor for the earth-moon force. 
* I adjusted the "force vector arrow scale" factors so defaults on the "Model" screen would be half a grid cell
* Eliminated all time scale factors (values now match the "To Scale" screen)

This was a non-trivial change and should be carefully reviewed.

@arouinfar or @kathy-phet can you please take a look?

**Comment by arouinfar** [comment]
@samreid things are generally looking okay -- most data the orbital periods are correct. I'd like to update the following values in the model. Most of the changes are in the last digit or two, except for `MOON_PERIGREE` which was significantly off.

| Constant | Value | Source |
| --- | --- | --- |
| SUN_RADIUS | 6.957E8 | https://nssdc.gsfc.nasa.gov/planetary/factsheet/sunfact.html |
| EARTH_MASS | 5.9724E24 | https://nssdc.gsfc.nasa.gov/planetary/factsheet/earthfact.html |
| EARTH_PERIHELION | 147098074 | http://astropixels.com/ephemeris/perap2001.html |
| MOON_RADIUS | 1727.4E3 | https://nssdc.gsfc.nasa.gov/planetary/factsheet/moonfact.html |
| MOON_MASS | 7.346E22 | https://nssdc.gsfc.nasa.gov/planetary/factsheet/moonfact.html |
| MOON_PERIGEE | 363300E3 | https://nssdc.gsfc.nasa.gov/planetary/factsheet/moonfact.html |

Something is still off withe Earth/Moon system.  The initial value of `toScaleScreen.model.planetMoonScene.planetMoonPair.distanceProperty` is `MOON_PERIGREE`.  However, after watching one complete orbit in the sim, it's clear that the starting position **isn't** the moon's closest distance. 
![image](https://user-images.githubusercontent.com/8419308/80650853-fe23a180-8a31-11ea-8fda-01525f921c21.png)

The moon's orbital speed is also off.  `toScaleScreen.model.planetMoonScene.moon.speedProperty` varies 997-1019 m/s. According to [NASA](https://nssdc.gsfc.nasa.gov/planetary/factsheet/moonfact.html), the range is actually 970-1082 m/s. At perigree, the speed would be 1082 m/s.

Looking at SceneFactory.js, it looks like the orbital speed of the Moon is relative to the Earth, so I wasn't able to dig into this further.
```js
const MOON_EARTH_SPEED = -1.01E3;
const MOON_SPEED = MOON_EARTH_SPEED;
```

**Comment by samreid** [comment]
After making the changes in the preceding comment, the moon-earth orbit looks like this:

![image](https://user-images.githubusercontent.com/679486/82222276-4bcb5580-98de-11ea-82de-eb0b0a29f051.png)

I'll commit and send over to @arouinfar for review.

**Comment by samreid** [comment]
I revised the values in the code based on @arouinfar's comment, and showed the new moon-earth orbit above.  @arouinfar can you please review?

**Comment by arouinfar** [comment]
@samreid the moon's orbital period, orbit path, and the `planetMoonPair.distanceProperty` are looking good throughout. 

At scene startup, `moon.speedProperty` should be `MOON_SPEED_AT_PERIGEE` (1082 m/s), but it's showing up in studio as 1068.8531987178003 m/s. As a result, the speeds throughout the orbit are consistently off by ~10 m/s. Can you look into it?

**Comment by samreid** [comment]
The problem is that the moon-planet system is moving to the left at 13m/s.  There is code that zeroes out the momentum--even though the relative speed is correct.  If I set the earth speed to ```
const planetV = -MOON_SPEED_AT_PERIGEE * MOON_MASS / EARTH_MASS;
```
then we have the correct value of `1082 m/s` in studio but the incorrect relative speed because now the earth is moving.

**Comment by samreid** [comment]
After the above change, the "centering" code indicates these overall velocities for the 4 scenes:

star planet: 0, -0.09098198925046125
star planet moon: 0.00003996152760090435, -0.09210105639512679
planet moon: 0, 0
planet satellite: 0, -5.415579750184181e-16

**Comment by samreid** [comment]
Here is the arithmetic @arouinfar and I used to center the moon-planet system and keep the correct relative moon speed at perihelion:

![image](https://user-images.githubusercontent.com/679486/82355207-22362b00-99bf-11ea-8cc9-1f64ed59f98e.png)

**Comment by arouinfar** [comment]
@samreid this is all looking good in master.
